### PR TITLE
Fix queries running multiple times in application choices index

### PR DESCRIPTION
**Pull request by steventux**
## Context

Obtaining the count of applications and then checking whether we had applications to render was causing duplicate queries to run in the `/provider/applications` action and views.

## Changes proposed in this pull request

Use Kaminari's `#total_count` method after `#load` to reduce the number of queries. [Kaminari will check to see if records have been loaded when determining the total count of records](https://github.com/kaminari/kaminari/blob/00657f252244413b512349f4de20c63ee478baa5/kaminari-activerecord/lib/kaminari/activerecord/active_record_relation_methods.rb#L25), this allows us to make one query for the page of applications _if_ the applications count is less than the page size. This won't always be the case and inevitably we'll see a query with limit and offset (fetching a page of results) _and_ a count query for total record count. 
This seemed to be the best compromise as it removes the EXISTS query being issued by the call to `any?` in the view. In some cases, (eg. pages with refined results from filtering) it will only issue one query for the results as the results are contained in one page. In most cases there will be a paginating query and a count query for applications.
 

## Guidance to review

Currently we perform 3 `ApplicationChoice` db queries when rendering the applications index (count, exists? and load). These queries can be seen in the development log.
This PR should reduce this so that for 1 page of results there should only be 1 `ApplicationChoice` db query and for more than 1 page of results there should be 2 queries (pagination and count).

## Link to Trello card

https://trello.com/c/d1lsWdCD/3681-fix-queries-running-multiple-times-in-applicationchoicesindex

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/queries/get_application_choices_for_providers.rb [hunk 1]
```diff
@@ -4,9 +4,9 @@ class GetApplicationChoicesForProviders
     :current_course_option,
     :provider,
     :site,
-    application_form: %i[candidate application_references application_work_experiences application_volunteering_experiences application_qualifications application_work_history_breaks],
-    course: %i[provider accredited_provider],
-    course_option: [{ course: %i[provider] }, :site],
+    { application_form: %i[candidate application_references application_work_experiences application_volunteering_experiences application_qualifications application_work_history_breaks] },
```
Comment (benswannack) [inline app/queries/get_application_choices_for_providers.rb:7]: What do the extra brackets do here?

Comment (steventux) [reply app/queries/get_application_choices_for_providers.rb:7]: I got a Rubocop warning https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/HashAsLastArrayItem so I did what I was told.

Comment (benswannack) [reply app/queries/get_application_choices_for_providers.rb:7]: Ah, can we just have one set of brackets around all three lines then?

**Review by benswannack — APPROVED** [review]
This all looks sensible 👍

File: app/queries/get_application_choices_for_providers.rb (hunk 1 above)
Comment (steventux) [reply app/queries/get_application_choices_for_providers.rb]: :+1: https://github.com/DFE-Digital/apply-for-teacher-training/pull/4752/commits/b77fb872384727967d77f65057f9ae7f1e8f63bd

**Review by cwrw — APPROVED** [review]
looks good, now down to 2 queries. Couldn't find a trivial way for Kaminari total count

**Pull request merged**